lib/compat.php

```diff
@@ -132,10 +132,29 @@ function gutenberg_ensure_wp_api_request() {
  */
 function gutenberg_disable_editor_settings_wpautop( $settings ) {
 	$post = get_post();
-	if ( ! isset( $settings['wpautop'] ) ) {
-		$settings['wpautop'] = ! ( is_object( $post ) && gutenberg_post_has_blocks( $post->ID ) );
+	if ( is_object( $post ) && gutenberg_post_has_blocks( $post->ID ) ) {
+		$settings['wpautop'] = false;
 	}
 
 	return $settings;
 }
 add_filter( 'wp_editor_settings', 'gutenberg_disable_editor_settings_wpautop' );
+
+/**
+ * WPCOM markdown support causes issues when saving a Gutenberg post by
+ * stripping out the <p> tags. This adds a filter prior to saving the post via
+ * REST API to disable markdown support. Markdown support most typically
+ * provided by Jetpack.
+ *
+ * @since 1.3.0
+ *
+ * @param  array $post      Post object which contains content to check for block.
+ * @return array $post      Post object.
+ */
+function gutenberg_remove_wpcom_markdown_support( $post ) {
```

Comment:
Shouldn't this live in the plugin that adds `wpcom-markdown` to posts?

Reply:
@aaronjorbin it does makes sense to include in Jetpack. However, there are also other plugins, for example <a href="https://wordpress.org/plugins/jetpack-markdown/">JP Markdown</a> which include it.

Reply:
@mkaz What about the [other plugins](https://wordpress.org/plugins/tags/markdown/) that add markdown in a different way? Should the code for turning off markdown in each of them live in Gutenberg or should Gutenberg provide the tools (such as the `content_has_blocks` function and ideally some documentation) to assist them?

Reply:
@aaronjorbin Gutenberg definitely should provide tools to assist, which this change does include one.

It's a bit of a catch-22, Gutenberg is still a plugin under development, trying to grow and get more people interested and gather feedback. If user wants to try it out and then their first experience is a broken, it's not a great way to gather feedback. So this fix disables a feature in one of the most popular plugins that makes for a poor Gutenberg experience.

As Gutenberg grows and gets closer to becoming the core editor, plugins will adopt the changes and include fixes themselves. However, as stated it is harder to grow if initial experience is rough.

Ideally, a plugin could provide a markdown block which would be awesome and any parsing conflict could also be avoided.

Reply:
I completely agree on the catch-22, but I think it also goes other ways. If Gutenberg is going to be in core, it can't be responsible for changes to plugins.  Either things need to be done in a fully backward compatible way, or education is needed for plugin maintainers to help them adjust. Playing favorites for popular plugins is going going to lead to unnecessary drama (what qualifies as a popular plugin?  Why was that plugin given preference over mine?), an increased and unnecessary maintenance burdon for core, and slower fixes for users.

Perhaps what's needed is some documentation and a documentation standard for things that can't be included in a core merge if that was to happen?